test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/TestAutoCreateSharedArchiveUpgrade.java

```diff
@@ -61,21 +72,33 @@ public class TestAutoCreateSharedArchiveUpgrade {
     private static String newJVM;
 
     public static void main(String[] args) throws Throwable {
-        setupJVMs();
-        doTest();
+        // Get OS and CPU type
+        String arch = props.getProperty("os.arch");
+        String os = getOsId();
+
+        // Earliest testable version is 19
+        int n = java.lang.Runtime.version().major() - 1;
+        for (int i = 19; i < n; i++) {
+            BOOT_JDK = fetchBootJDK(os, arch, i);
+            setupJVMs(os);
+            doTest();
+        }
```

Comment:
If fetchBootJDK() cannot connect to the artifactory, it will always returns the same path regardless of the version. In that case, we shouldn't call doTest() more than once.